android-components/components/feature/search/src/main/java/mozilla/components/feature/search/middleware/SearchMiddleware.kt

```diff
@@ -177,6 +183,18 @@ class SearchMiddleware(
         )
     }
 
+    private fun updateDisabledSearchEngineShortcutsIds(
+        action: SearchAction.UpdateDisabledSearchEngineShortcutsIdsAction,
+    ) = scope.launch {
+        val disabledIds = metadataStorage.getDisabledSearchEngineShortcutsIds()
```

Comment:
We have already loaded the most recent state from `loadSearchEngines` so we don't need to query the storage layer again. Instead, we can use the most version from the SearchState we have already fetched.

Reply:
implemented